source/unpromise.d.ts

```diff
@@ -0,0 +1,10 @@
+/**
+Unwraps `Promise` and returns lifted type.
+
```

Comment:
> Write about some real-world use-cases where it can be useful. (It can be hard sometimes for users to see where they would use something) - https://github.com/sindresorhus/type-fest/blob/master/.github/contributing.md#submitting-a-new-type